docs: update documentation and examples (#1408)
splunk/addonfactory-ucc-generator: docs/entity/validators.md

See the following example usage:

```json
{
  "type": "text",
  "label": "Email",
  "validators": [
      {
        "type": "email",
      }
  ],
  "field": "email"
}
```

### IPV4

No parameters are needed.

Internally, it checks the IPV4 address using this [regex](https://github.com/splunk/addonfactory-ucc-generator/blob/main/ui/src/constants/preDefinedRegex.ts).

See the following example usage:

```json
{
    "field": "testIpv4",
    "label": "Test Ipv4",
    "type": "text",
    "validators": [
        {
            "type": "ipv4"
        }
    ]
}
```

### Date

No parameters are needed.

It is validated if the field's value is a date in the [ISO 8601](https://www.w3.org/TR/1998/NOTE-datetime-19980827) format.
It uses the regex from [moment.js](https://github.com/moment/moment/blob/2.17.1/moment.js#L1980).

See the following example usage:

```json
{
    "field": "testDate",
    "label": "Test Date",
    "type": "text",
    "validators": [
        {
            "type": "date"
        }
    ]
}
```

### Combinations

You can also make a combinations of many validators by providing all of them in single array.

> See the following example usage of validators :

Reviewer: You can combine multiple validators in a single array.

Example:

> You can also make a combinations of many validators by providing all of them in single array.
> 
> See the following example usage of validators :

Reviewer: Maybe shorten it a bit

```suggestion
You can combine multiple validators in a single array.

Example:
```

```json
{

>     "field": "long_url",
>     "label": "Insert url at least 10 characters long",

Reviewer: ```suggestion
    "field": "https_url",
    "label": "HTTPS only URL",
```
